### PR TITLE
Redirect natively if no baseurl is set

### PR DESCRIPTION
**Pull request by Shadowghost**
Based on the code from #4546 but uses `System.Uri` to generate the proper redirect path.

**Changes**
Fix redirect failures by redirecting back to the index page.

----

Supersedes #4546

### CODE REVIEW

**Review by crobibero — COMMENTED** [review]
This removes the health check redirect

**Comment by Shadowghost** [comment]
I guess we need to ecplicitly exclude it. What about the metrics endpoint?

**Comment by crobibero** [comment]
The `/health` redirect is there for mostly automatic docker health checks, so that redirect should still exist. The metrics endpoint shouldn't automatically redirect, the user should just configure their custom tools correctly

**Review by crobibero — APPROVED** [review]

File: Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs [hunk 1]
```diff
@@ -45,36 +45,40 @@ public async Task Invoke(HttpContext httpContext, IServerConfigurationManager se
...
                 }
 
-                if (!startsWithBaseUrl
-                    || localPath.Length == baseUrlPrefix.Length
-                    // Local path is /baseUrl/
-                    || (localPath.Length == baseUrlPrefix.Length + 1 && localPath[^1] == '/'))
+                // Always redirect back to the default path if the base prefix is invalid or missing
+                _logger.LogDebug("Normalizing an URL at {LocalPath}", localPath);
+
+                if (string.IsNullOrWhiteSpace(baseUrlPrefix) || baseUrlPrefix[^1] != '/')
```
Comment (cvium) [inline Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs:71]: This is always true https://github.com/jellyfin/jellyfin/blob/90174f68e2b477e613d756d78cbfd89c2a33f30a/Jellyfin.Networking/Configuration/NetworkConfiguration.cs#L64

File: Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs [hunk 2]
```diff
@@ -45,36 +45,35 @@ public async Task Invoke(HttpContext httpContext, IServerConfigurationManager se
             var localPath = httpContext.Request.Path.ToString();
             var baseUrlPrefix = serverConfigurationManager.GetNetworkConfiguration().BaseUrl;
 
-            if (!string.IsNullOrEmpty(baseUrlPrefix))
+            if (string.IsNullOrEmpty(localPath)
+                || string.Equals(localPath, "/", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(localPath, "/web", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(localPath, "/web/", StringComparison.OrdinalIgnoreCase)
```
Comment (cvium) [inline Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs:51]: These seem very redundant given the `!localPath.StartsWith(baseUrlPrefix, StringComparison.OrdinalIgnoreCase)` check

**Review by cvium — APPROVED** [review]

**Pull request merged**